build: `--target deploy` failure on macOS

macOS 15.4
master @ 51166559808c3528f148b5c0c38cb4481e536dd8
brew installed Qt 6.9.0
```bash
cmake -B build -DBUILD_GUI=ON
cmake --build build --target deploy
[  0%] Generating bitcoin-build-info.h
[  0%] Built target generate_build_info
[  5%] Built target bitcoin_wallet
[ 13%] Built target leveldb
[ 15%] Built target bitcoin_cli
[ 15%] Built target bitcoinqt_autogen_timestamp_deps
[ 15%] Built target bitcoinqt_autogen
[ 51%] Built target bitcoinqt
[ 69%] Built target bitcoin_node
[ 69%] Built target bitcoin-qt_autogen_timestamp_deps
[ 69%] Built target bitcoin-qt_autogen
[ 71%] Built target crc32c
[ 73%] Built target minisketch
[ 73%] Built target bitcoin_clientversion
[ 75%] Built target bitcoin_consensus
[ 84%] Built target bitcoin_common
[ 86%] Built target univalue
[ 86%] Built target secp256k1_precomputed
[ 88%] Built target secp256k1
[ 92%] Built target bitcoin_crypto
[ 98%] Built target bitcoin_util
[ 98%] Built target bitcoin_zmq
[100%] Built target bitcoin-qt
[100%] Generating Bitcoin-Core.zip
+ Removing existing dist folder +
+ Copying source bundle +
+ Deploying frameworks +
Processing QtWidgets.framework ...
<snip>
libglib-2.0.0.dylib
/Library/Developer/CommandLineTools/usr/bin/install_name_tool: warning: changes being made to the file will invalidate the code signature in: dist/Bitcoin-Qt.app/Contents/Frameworks/libglib-2.0.0.dylib
/Library/Developer/CommandLineTools/usr/bin/install_name_tool: warning: changes being made to the file will invalidate the code signature in: dist/Bitcoin-Qt.app/Contents/Frameworks/libglib-2.0.0.dylib
/Library/Developer/CommandLineTools/usr/bin/install_name_tool: warning: changes being made to the file will invalidate the code signature in: dist/Bitcoin-Qt.app/Contents/Frameworks/libglib-2.0.0.dylib
Processing QtDBus.framework ...
Error: No directory at @rpath/QtDBus.framework
gmake[3]: *** [CMakeFiles/deploydir.dir/build.make:73: Bitcoin-Core.zip] Error 1
gmake[2]: *** [CMakeFiles/Makefile2:663: CMakeFiles/deploydir.dir/all] Error 2
gmake[1]: *** [CMakeFiles/Makefile2:702: CMakeFiles/deploy.dir/rule] Error 2
gmake: *** [Makefile:225: deploy] Error 2
```

@i-am-yuvi 

> ```
> brew list --versions qt@5
> qt@5 5.15.13_1
> ```
> 
>  > How could this possibly work fine, given that qt@6 is required?
> 
> Is it? I thought that qt@6 is the recommended version.

Are you sure you are building the recent master branch?

On my machine:
```
% git rev-parse HEAD
cdc32994feadf3f15df3cfac5baae36b4b011462
% brew list | grep qt
qt@5
% rm -rf build && cmake -B build -DBUILD_GUI=ON
-- The CXX compiler identification is AppleClang 16.0.0.16000026
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Setting build type to "RelWithDebInfo" as none was specified
-- Performing Test CXX_SUPPORTS__WERROR
-- Performing Test CXX_SUPPORTS__WERROR - Success
-- Performing Test CXX_SUPPORTS__G3
-- Performing Test CXX_SUPPORTS__G3 - Success
-- Performing Test LINKER_SUPPORTS__G3
-- Performing Test LINKER_SUPPORTS__G3 - Success
-- Performing Test CXX_SUPPORTS__FTRAPV
-- Performing Test CXX_SUPPORTS__FTRAPV - Success
-- Performing Test LINKER_SUPPORTS__FTRAPV
-- Performing Test LINKER_SUPPORTS__FTRAPV - Success
-- Found SQLite3: /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include (found suitable version "3.43.2", minimum required is "3.7.17")
-- Found QRencode: /usr/local/lib/libqrencode.dylib
-- Could NOT find Qt6 (missing: Qt6_DIR)
CMake Error at /usr/local/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:227 (message):
  Could NOT find Qt (missing: Qt6_DIR) (Required is at least version "6.2")
Call Stack (most recent call first):
  /usr/local/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:591 (_FPHSA_FAILURE_MESSAGE)
  cmake/module/FindQt.cmake:38 (find_package_handle_standard_args)
  CMakeLists.txt:178 (find_package)


-- Configuring incomplete, errors occurred!
```

--- a/ci/test/00_setup_env_mac_native.sh
+++ b/ci/test/00_setup_env_mac_native.sh
@@ -9,7 +9,7 @@ export LC_ALL=C.UTF-8
 # Homebrew's python@3.12 is marked as externally managed (PEP 668).
 # Therefore, `--break-system-packages` is needed.
 export PIP_PACKAGES="--break-system-packages zmq"
-export GOAL="install"
+export GOAL="install deploy"
 export CMAKE_GENERATOR="Ninja"
 export BITCOIN_CONFIG="-DBUILD_GUI=ON -DWITH_ZMQ=ON -DREDUCE_EXPORTS=ON"
 export CI_OS_NAME="macos"

--- a/contrib/macdeploy/macdeployqtplus
+++ b/contrib/macdeploy/macdeployqtplus
@@ -181,7 +181,7 @@ class DeploymentInfo(object):
                     return True
         return False
 
-def getFrameworks(binaryPath: str, verbose: int) -> list[FrameworkInfo]:
+def getFrameworks(binaryPath: str, verbose: int, rpath: str = '') -> list[FrameworkInfo]:
     objdump = os.getenv("OBJDUMP", "objdump")
     if verbose:
         print(f"Inspecting with {objdump}: {binaryPath}")
@@ -195,17 +195,19 @@ def getFrameworks(binaryPath: str, verbose: int) -> list[FrameworkInfo]:
     lines.pop(0) # First line is the inspected binary
     if ".framework" in binaryPath or binaryPath.endswith(".dylib"):
         lines.pop(0) # Frameworks and dylibs list themselves as a dependency.
-    
+
     libraries = []
     for line in lines:
         line = line.replace("@loader_path", os.path.dirname(binaryPath))
+        if rpath:
+            line = line.replace("@rpath", rpath)
         info = FrameworkInfo.fromLibraryLine(line.strip())
         if info is not None:
             if verbose:
                 print("Found framework:")
                 print(info)
             libraries.append(info)
-    
+
     return libraries
 
 def runInstallNameTool(action: str, *args):
@@ -318,7 +320,7 @@ def deployFrameworks(frameworks: list[FrameworkInfo], bundlePath: str, binaryPat
         # install_name_tool it a new id.
         changeIdentification(framework.deployedInstallName, deployedBinaryPath, verbose)
         # Check for framework dependencies
-        dependencies = getFrameworks(deployedBinaryPath, verbose)
+        dependencies = getFrameworks(deployedBinaryPath, verbose, rpath=framework.frameworkDirectory)
         
         for dependency in dependencies:
             changeInstallName(dependency.installName, dependency.deployedInstallName, deployedBinaryPath, verbose)

--- a/contrib/macdeploy/macdeployqtplus
+++ b/contrib/macdeploy/macdeployqtplus
@@ -157,20 +157,19 @@ class DeploymentInfo(object):
         self.qtPath = None
         self.pluginPath = None
         self.deployedFrameworks = []
-    
+
     def detectQtPath(self, frameworkDirectory: str):
         parentDir = os.path.dirname(frameworkDirectory)
-        if os.path.exists(os.path.join(parentDir, "translations")):
-            # Classic layout, e.g. "/usr/local/Trolltech/Qt-4.x.x"
+        if os.path.exists(os.path.join(parentDir, "share", "qt", "translations")):
             self.qtPath = parentDir
         else:
             self.qtPath = os.getenv("QTDIR", None)
 
         if self.qtPath is not None:
-            pluginPath = os.path.join(self.qtPath, "plugins")
+            pluginPath = os.path.join(self.qtPath, "share", "qt", "plugins")
             if os.path.exists(pluginPath):
                 self.pluginPath = pluginPath
-    
+
     def usesFramework(self, name: str) -> bool:
         for framework in self.deployedFrameworks:
             if framework.endswith(".framework"):

